test/test.go

```diff
@@ -37,6 +48,35 @@ func WriteConfig(t *testing.T, path string, cfg *config.Config) {
 	if err = encoder.Encode(cfg); err != nil {
 		t.Fatalf("failed to write to config file: %v", err)
 	}
+
+	// Ensure the modtime of the config file always increases
+	// (even if this requires setting it to the future!)
+	// If we don't do this, we end up with flaky tests that behave differently
+	// depending on if they run quickly enough for the modtime to stay constant throughout the test.
+	newInfo, err := os.Lstat(path)
+	if err != nil {
+		t.Fatalf("failed to create a new config file: %v", err)
+	}
+	// Note: we're comparing `Unix()` (which is only 1 second granularity) for consistency with
+	// `walk.go::formatSignature`.
+	if oldInfo != nil && oldInfo.ModTime().Unix() == newInfo.ModTime().Unix() {
+		// No change to atime.
+		sysStat, ok := newInfo.Sys().(*syscall.Stat_t)
+		if !ok {
+			t.Fatalf("failed to cast stat Sys")
+		}
+
+		atime := sysStat.Atim
```

Comment:
This fails on darwin:

```
treefmt> test/test.go:69:20: sysStat.Atim undefined (type *syscall.Stat_t has no field or method Atim)
```

I don't even want to know the atime, I just want to change the mtime, but `Lutimes` wants both an atime and a mtime.

Easiest fix might be to just set atime to the current timestamp, as we don't really care about atime in treefmt.